### PR TITLE
add kmesh-daemon dump command

### PR DESCRIPTION
**Pull request by Okabe-Rintarou-0**
**What type of PR is this?**
enhancement

**What this PR does / why we need it**:
add kmesh-daemon dump command
截屏2024-06-17 19 37 21

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: daemon/manager/dump/dump.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/spf13/cobra"
+
+	"kmesh.net/kmesh/pkg/status"
+)
+
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dump",
+		Short: "Dump xds workloads",
+		Args:  cobra.ExactArgs(1),
```
Comment (LiZhenCheng9527) [inline daemon/manager/dump/dump.go:18]: Can you add an example on how to use it?

**Review by LiZhenCheng9527 — COMMENTED** [review]
Add CopyRight

File: daemon/manager/dump/dump.go (hunk 1 above)
Comment (Okabe-Rintarou-0) [reply daemon/manager/dump/dump.go:18]: Provides a shortcut for curl localhost:15200/debug/config_dump/workload
image

Comment (Okabe-Rintarou-0) [reply daemon/manager/dump/dump.go:34]: > Can you add an example on how to use it?

Sure

File: daemon/manager/dump/dump.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+	"github.com/spf13/cobra"
+
+	"kmesh.net/kmesh/pkg/status"
+)
+
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dump",
+		Short: "Dump xds workloads",
```
Comment (hzxuzhonghu) [inline daemon/manager/dump/dump.go:33]: I can see it can dump not only workloads.

Comment (Okabe-Rintarou-0) [reply daemon/manager/dump/dump.go:33]: I want to provide a shortcut for dump config of both ads and workload mode

what about this?

```go
func NewCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "dump",
		Short: "Dump config of ads or workload mode",
		Example: `Ads mode:
		kmesh-daemon dump ads
	  
	  Workload mode:
		kmesh-daemon dump workload`,
		Args: cobra.ExactArgs(1),
		Run: func(cmd *cobra.Command, args []string) {
			_ = RunDump(cmd, args)
		},
	}
	return cmd
}
```

Comment (hzxuzhonghu) [reply daemon/manager/dump/dump.go:33]: lgtm

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

In the long run, a tool like `istioctl proxy-config` is simpler to use.

**Pull request merged**